src/psyclone/tests/psyGen_test.py

```diff
@@ -2226,3 +2226,41 @@ def test_walk():
 
     binary_op_list = invoke.schedule.walk(BinaryOperation, Kern)
     assert not binary_op_list
+
+
+def test_split_consecutive1():
+    '''Tests the split_consecutive functionality for imperfect_nest.f90.'''
+
+    _, invoke = get_invoke("imperfect_nest.f90", "nemo", 0)
+
+    # There is only one pair of adjacent loops, the rest stand alone
+    for i, block in enumerate(invoke.schedule.split_consecutive(Loop)):
+        assert len(block) == (2 if i == 2 else 1)
+
+    # There is only one block of consecutive assignments, the rest stand alone
+    for i, block in enumerate(invoke.schedule.split_consecutive(Assignment)):
+        assert len(block) == (4 if i == 3 else 1)
+
+
+def test_split_consecutive2():
+    '''Tests the split_consecutive functionality for
+    explicit_do_two_loops.f90.'''
+
+    _, invoke = get_invoke("explicit_do_two_loops.f90", "nemo", 0)
```

Comment:
Again, when you move this test, please could you re-write to use `fortran_reader`.

Reply:
Done. See f65ee7f25436497e6a991a0c79f3047f154d93a1.